services/clojars/clojars-base.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const Joi = require('joi')
+const { version: versionColor } = require('../color-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService, InvalidResponse } = require('..')
+
+const clojarsSchema = Joi.object({
+  downloads: nonNegativeInteger,
+  latest_version: Joi.string().required(),
+  latest_release: Joi.string().required(),
+}).required()
+
+class BaseClojarsService extends BaseJsonService {
+  async fetch({ clojar }) {
+    // Clojars API Doc: https://github.com/clojars/clojars-web/wiki/Data
+    const url = `https://clojars.org/api/artifacts/${clojar}`
+    return this._requestJson({
+      url,
+      schema: clojarsSchema,
+    })
+  }
+}
+
+class BaseClojarsVersionService extends BaseClojarsService {
+  static render({ clojar, version }) {
+    return {
+      message: `[${clojar} "${version}"]`,
+      color: versionColor(version),
+    }
+  }
+
+  static transform(json, version) {
+    if (version === 'release') {
```

Comment:
the transition to the new API has eliminated a lot of the repetitive code that existed before. It seems that with the new API now, the snapshot badge is grabbing the `latest_version` field, and the release version badge is grabbing the `latest_release` field (no sharing/duplication).

It might be easier to just delete of this function now, and then updating the two service classes to grab their desired field in their respective `handle` functions. For example the release version service could become:

```js
  async handle({ clojar }) {
    const json = await this.fetch({ clojar })
    return this.constructor.render({ clojar, version: json.latest_release })
  }
```

Sorry, didn't mean to flip flop on `transform` 😄  but the code required for the transform process was dramatically reduced 👍 With the current service implementations the `else` block here can't/won't be hit otherwise

Reply:
removed transform as it's now not needed.